### PR TITLE
lesson7 Task2

### PR DESCRIPTION
**Pull request by xRuTeRx**

### CODE REVIEW

File: lesson7/task2/cmd/server/main.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import (
+	"log"
+	"net/http"
+
+	"github.com/xruterx/golang/lesson7/task2/handlers"
+)
+
+func main() {
+	addr := "localhost:8081"
+	endpoint := "/rpc"
```
Comment (denislituev) [inline lesson7/task2/cmd/server/main.go:12]: do not need here separate variable for endpoint

File: lesson7/task2/handlers/handle.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	returnOK(w, req.Id, resp)
+}
+
+func parseRequest(r *http.Request) (*Request, error) {
+	body := r.Body
+	defer body.Close()
+	bodyData, err := io.ReadAll(body)
+	if err != nil {
+		log.Print(err.Error())
+		return nil, errors.New("failed to read request body")
```
Comment (denislituev) [inline lesson7/task2/handlers/handle.go:46]: It is good GO style to have a block of constants in the top of the file with all errors messages

File: lesson7/task2/handlers/handle.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	}
+	req := &Request{}
+
+	if err = json.Unmarshal(bodyData, req); err != nil {
+		log.Print(err.Error())
+		return nil, errors.New("request format is wrong")
```
Comment (denislituev) [inline lesson7/task2/handlers/handle.go:52]: also could move string literal to const block

File: lesson7/task2/handlers/handle.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+func handle(req Request) (interface{}, error) {
+	handlers := map[string]func(message json.RawMessage) (interface{}, error){
+		"register": methods.Register,
+		"list":     methods.List,
+	}
+	h, ok := handlers[req.Method]
+	if !ok {
+		return nil, errors.New(fmt.Sprintf("method %s is not supported", req.Method))
```
Comment (denislituev) [inline lesson7/task2/handlers/handle.go:65]: also could move string literal to const block

File: lesson7/task2/handlers/methods/list.go [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+	"errors"
+)
+
+type listParams struct {
+}
+
+func List(params json.RawMessage) (interface{}, error) {
+	p := listParams{}
+	if err := json.Unmarshal(params, &p); err != nil {
+		return "", errors.New("wrong params sent, expected {name:string}")
```
Comment (denislituev) [inline lesson7/task2/handlers/methods/list.go:14]: same here

File: lesson7/task2/cmd/client/main.go [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+	Params json.RawMessage `json:"params"`
+	Id     int64           `json:"id"`
+}
+type jsonParams struct {
+	Name string `json:"name"`
+}
+
+func main() {
+
+	const port = 8081
```
Comment (denislituev) [inline lesson7/task2/cmd/client/main.go:28]: this const could be declared out of main()

File: lesson7/task2/cmd/client/main.go [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+	address := fmt.Sprintf("http://localhost:%d/rpc", port)
```
Comment (denislituev) [inline lesson7/task2/cmd/client/main.go:29]: also make this string as const - this will be look better

**Review by denislituev — CHANGES_REQUESTED** [review]
good job, please fix comments and fix string errors even if I missed it

**Review by nikonok — APPROVED** [review]

**Review by denislituev — APPROVED** [review]

**Pull request merged**